pkg/ingester/ingester.go

```diff
@@ -3764,6 +3763,36 @@ func (i *Ingester) ShutdownHandler(w http.ResponseWriter, _ *http.Request) {
 	w.WriteHeader(http.StatusNoContent)
 }
 
+// startReadRequest tries to start a read request and returns a readRequestState object
+// representing the state of a possibly started read request. If it wasn't possible to
+// start the read request, the causing error is returned.
+func (i *Ingester) startReadRequest() (*readRequestState, error) {
+	start := time.Now()
+	acquiredCircuitBreakerPermit, err := i.circuitBreaker.tryAcquirePermit()
+	if err != nil {
+		return nil, err
+	}
+	err = i.checkAvailableForRead()
+	if err != nil {
+		return nil, err
+	}
+	err = i.checkReadOverloaded()
+	if err != nil {
+		return nil, err
+	}
+	return &readRequestState{requestStart: start, acquiredCircuitBreakerPermit: acquiredCircuitBreakerPermit}, err
+}
+
+// finishReadRequest finishes a previously started read request, whose state is described
+// by the passed readRequestState object. If an error occurred during the read request
+// execution, finishReadRequest returns it.
+func (i *Ingester) finishReadRequest(st *readRequestState, err error) error {
+	if st != nil && st.acquiredCircuitBreakerPermit {
+		_ = i.circuitBreaker.finishReadRequest(time.Since(st.requestStart), err)
```

Comment:
I see that the return value isn't used here, same for `finishPushRequest`, can we simplify the code by removing these return values completely?

Reply:
I have changed this. Thank you